selfservice/hook/web_hook.go

```diff
@@ -307,9 +346,48 @@ func doHttpCall(method string, url string, as AuthStrategy, body io.Reader) erro
 
 	if err != nil {
 		return err
-	} else if resp.StatusCode >= 400 {
+	} else if resp.StatusCode >= http.StatusBadRequest {
+		if interrupt {
+			if err = parseResponse(resp); err != nil {
+				return err
+			}
+		}
 		return fmt.Errorf("web hook failed with status code %v", resp.StatusCode)
 	}
 
 	return nil
 }
+
+func parseResponse(resp *http.Response) error {
+	if resp == nil {
+		return fmt.Errorf("empty response provided from the webhook")
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "could not read response body")
+	}
+	defer func(Body io.ReadCloser) {
```

Comment:
Why we are doing it by `func` here, and not just `defer resp.Body.Close()`, do we plan to check for error?

Reply:
It is not that obvious since we can't do it properly in the closure.
This should be OK f063c6f1